src/gtfs/read.rs

```diff
@@ -879,6 +881,155 @@ pub fn set_dataset_validity_period(
+    let mut trip_id_sequence: HashMap<String, u8> = HashMap::new();
+    let mut vehicle_journeys = collections.vehicle_journeys.take();
+    for frequency in &gtfs_frequencies {
+        if frequency.start_time == frequency.end_time {
+            warn!(
+                "frequency for trip {:?} has same start and end time",
+                frequency.trip_id
+            );
+            continue;
+        }
+        let datetime_estimated = match frequency.exact_times {
+            FrequencyExactTime::Exact => false,
+            FrequencyExactTime::Inexact => true,
+        };
+        let corresponding_vj = skip_fail!(
+            vehicle_journeys
+                .iter()
+                .find(|&vj| vj.id == frequency.trip_id)
+                .cloned()
+                .ok_or_else(|| format_err!(
+                    "frequency mapped to an unexisting trip {:?}",
+                    frequency.trip_id
+                ))
+        );
+        let mut codes = corresponding_vj.codes.clone();
+        codes.insert(("source".to_string(), frequency.trip_id.clone()));
+        let mut start_time = frequency.start_time;
+        if corresponding_vj.stop_times.is_empty() {
+            warn!(
+                "frequency mapped to trip {:?} with no stop_times",
+                frequency.trip_id
+            );
+            continue;
+        }
+        let mut arrival_time_delta = corresponding_vj
+            .stop_times
+            .iter()
+            .min()
+            .unwrap()
+            .arrival_time;
+        while start_time < frequency.end_time {
+            let mut stop_times = vec![];
+            trip_id_sequence
+                .entry(frequency.trip_id.clone())
+                .and_modify(|counter| *counter += 1)
+                .or_insert(0);
+            let generated_trip_id = format!(
+                "{}-{}",
+                frequency.trip_id, trip_id_sequence[&frequency.trip_id]
+            );
+            // the following handles generated trip starting after midnight, we need to generate a
+            // new service in case the next day is not covered
+            let service_id = if start_time.hours() >= 24 {
+                arrival_time_delta = arrival_time_delta + Time::new(24, 0, 0);
+                let service = collections
+                    .calendars
+                    .get(&corresponding_vj.service_id)
+                    .cloned()
+                    .unwrap();
+                let new_dates: BTreeSet<_> = service
+                    .dates
+                    .iter()
+                    .map(|d| *d + chrono::Duration::days(1))
+                    .collect();
+                let new_service_id = format!("{}:{}", service.id, generated_trip_id);
+                let new_service = objects::Calendar {
+                    id: new_service_id.clone(),
+                    dates: new_dates,
+                };
+                collections.calendars.push(new_service)?;
```

Comment:
It will generate a lot of useless services: one for each instance another day that the original day.

Reply:
@prhod i talked about it with @papailio what do you think?

Reply:
updated i generate less services, and used format "service:3:+2days" are u all ok?

Reply:
I don't have opinion on the id generation, but having one service by shift resolve my comment.